code/game/objects/items/inspector.dm

```diff
@@ -93,6 +97,49 @@
 	else
 		. += "\The [cell] is firmly in place. [span_info("Ctrl-click with an empty hand to remove it.")]"
 
+/obj/item/inspector/interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
+	. = ..()
+	if(!interacting_with.can_interact(user, TRUE))
+		return
+	if(cell_cover_open)
+		balloon_alert(user, "close cover first")
+		return
+	if(!cell || !cell.use(INSPECTOR_ENERGY_USAGE_LOW))
+		balloon_alert(user, "check cell")
+		return
+	if(!isitem(interacting_with))
+		return
+	var/obj/item/contraband_item = interacting_with
+	var/contraband_status = contraband_item.is_contraband()
+	if((!contraband_status && scans_correctly) || (contraband_status && !scans_correctly))
+		playsound(src, 'sound/machines/ping.ogg', 20)
+		balloon_alert(user, "clear")
+		return
+
+	playsound(src, 'sound/machines/uplinkerror.ogg', 40)
+	balloon_alert(user, "contraband detected")
```

Comment:
Remember to return `ITEM_INTERACT_SUCCESS` on successful interactions, `ITEM_INTERACT_BLOCKING` if it failed and you don't want the attack chain to continue, or `NONE` to let them thwack the object